### PR TITLE
Make RBAC comptaible with multiple instances

### PR DESCRIPTION
**Pull request by jynolen**
### What this PR does / why we need it:
THIS PR Change the naming of RBAC Object created by the operator.
Today RBAC Object does not support multiple instances of FluentBit making secondary agent unable to query k8s apis

### Does this PR introduced a user-facing change?
None, except multiple clusterrolebinding / rolebinding creation during reconciliation

### CODE REVIEW

**Review by wenchajun — APPROVED** [review]

**Comment by benjaminhuo** [comment]
@wanjunlei would you help to review this? Does it necessary for each fluentbit instance to have separate RBAC objects?

File: pkg/operator/rbac.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func MakeRBACObjects(name, namespace, component string, additionalRules []rbacv1
 func MakeScopedRBACObjects(fbName, fbNamespace string, saAnnotations map[string]string) (*rbacv1.Role, *corev1.ServiceAccount, *rbacv1.RoleBinding) {
 	r := rbacv1.Role{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "kubesphere:fluent",
+			Name:      "fluent-operator:fluent",
```
Comment (wanjunlei) [inline pkg/operator/rbac.go:62]: `fluent:fluent-operator` is more appropriate.

File: pkg/operator/rbac.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func MakeRBACObjects(name, namespace, component string, additionalRules []rbacv1
 
 	crb := rbacv1.ClusterRoleBinding{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: rbacName,
+			Name: fmt.Sprintf("%s-%s", component, name),
```
Comment (wanjunlei) [inline pkg/operator/rbac.go:40]: The clusterrolebinding needs a `fluent-operator` prefix too.

**Review by wanjunlei — CHANGES_REQUESTED** [review]

File: pkg/operator/rbac.go (hunk 2 above)
Comment (benjaminhuo) [reply pkg/operator/rbac.go:40]: > The clusterrolebinding needs a `fluent-operator` prefix too.

Agree, we can use code like below?
```suggestion
			Name: fmt.Sprintf("%s-%s", rbacName, name),
```

File: pkg/operator/rbac.go (hunk 1 above)
Comment (benjaminhuo) [reply pkg/operator/rbac.go:62]: agree

File: pkg/operator/rbac.go [hunk 3]
```diff
@@ -94,7 +94,7 @@ func MakeScopedRBACObjects(fbName, fbNamespace string, saAnnotations map[string]
 		RoleRef: rbacv1.RoleRef{
 			APIGroup: rbacv1.GroupName,
 			Kind:     "Role",
-			Name:     "kubesphere:fluent",
+			Name:     "fluent-operator:fluent",
```
Comment (benjaminhuo) [inline pkg/operator/rbac.go:97]:
```suggestion
			Name:     "fluent:fluent-operator",
```

File: pkg/operator/rbac.go [hunk 4]
```diff
@@ -81,7 +81,7 @@ func MakeScopedRBACObjects(fbName, fbNamespace string, saAnnotations map[string]
 
 	rb := rbacv1.RoleBinding{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "kubesphere:fluent",
+			Name:      fmt.Sprintf("fluent-operator-%s", fbName),
```
Comment (benjaminhuo) [inline pkg/operator/rbac.go:84]: Maybe we can follow MakeRBACObjects' convention:
```suggestion
			Name:      fmt.Sprintf("fluent-operator-fluent-bit-%s", fbName),
```

**Review by jynolen — COMMENTED** [review]
Approved changes

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**